src/Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs

```diff
@@ -100,31 +121,71 @@ public BlockPullerBehavior(IBlockPuller blockPuller, IInitialBlockDownloadState
         }
 
         /// <inheritdoc/>
-        public void AddSample(double delaySeconds)
+        public void AddSample(long blockSizeBytes, double delaySinceRequestedSeconds)
         {
-            this.logger.LogTrace("({0}:{1})", nameof(delaySeconds), delaySeconds);
+            this.logger.LogTrace("({0}:{1},{2}:{3})", nameof(blockSizeBytes), blockSizeBytes, nameof(delaySinceRequestedSeconds), delaySinceRequestedSeconds);
+
+            double adjustedDelay;
+
+            if (this.lastDeliveryTime == null)
```

Comment:
Brackets around please

Reply:
Even better, you can simplify by initialising the values on line 128 with this and then apply adjustment in case last delivery time is available

Reply:
fixed